test/provisioner/common.go

```diff
@@ -0,0 +1,62 @@
+package provisioner
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"strings"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+	"sigs.k8s.io/e2e-framework/pkg/envconf"
+)
+
+type patchLabel struct {
+	Op    string            `json:"op"`
+	Path  string            `json:"path"`
+	Value map[string]string `json:"value"`
+}
+
+func AddNodeRoleWorkerLabel(ctx context.Context, clusterName string, cfg *envconf.Config) error {
+	fmt.Printf("Adding worker label to nodes belonging to: %s\n", clusterName)
+
+	config, err := clientcmd.BuildConfigFromFlags("", cfg.KubeconfigFile())
```

Comment:
@bookinabox I tested this PR, it is working. I've only a few comments to the code.

First, what if you get the client object like (https://pkg.go.dev/sigs.k8s.io/e2e-framework@v0.1.0/pkg/envconf#Config.NewClient):
```
	client, err := cfg.NewClient()
	if err != nil {
		return err
	}
```

Reply:
Well, the clientset and the client returned from that function are different types. I don't know why there are different ways to do the same thing/what the pros/cons are but I will try to do it this way. I imagine there will be some major changes required to do this though.

Reply:
Is there a specific reason why this way would be better? I'm having a little trouble getting the patch operation to work with resources (I'm sure there's an easy fix, I'm confused which k8s.object I need to pass into .Patch) and I'm not sure what the benefit to doing the above way would be.

Reply:
I saw that you managed to use the cfg.NewClient() which returns a `kclient` (https://github.com/kubernetes-sigs/e2e-framework/tree/main/klient), however I still own you some answers.

The e2e-framework provides some facilities over the standard k8s api. For example, when you start the test suite you can set the k8s namespace resources will be created (and destroyed on the end). The `kcliet` wraps that information so that you don't need to pass the namespace to every call of, for example, `Resources()`. The `kclient` has also special `wait` routines for resources. And so on.

There are other reasons to use 'kclient' instead of 'clientset`. For example, avoid dependence on another package, avoid code fragmentation by leveraging a single API thoughtout the code base. And so on.